backend/ops/HostInteropNode.cpp

```diff
@@ -1,108 +1,187 @@
-#include "../codegen.h"  
+#include "../codegen.h"
 
 using namespace std;
 using namespace llvm;
 
-// TODO: This is only a band-aid implementation!
-
 TypedValue CodeGenerator::codegen(const Node &node, const HostInteropNode &subnode, const ObjectTypeSet &typeRestrictions) {
-  auto methodName = subnode.morf();
-  string className = "vector"; // TODO: How to get that value at runtime?
```

Comment:
Like this:
`auto retVal = dynamicString(string("vector").c_str());
  dynamicRetain(retVal);  
`
This way the string is created during compilation not execution, only retain is during execution (see dynamicString for details, it is calling the runtime function directly).